integrations/tensorflow/e2e/math/math_test.py

```diff
@@ -0,0 +1,561 @@
+import collections
+import os
+from typing import Any, Dict, Sequence, Type, Union
+
+from absl import app
+from absl import flags
+import numpy as np
+from pyiree.tf.support import tf_test_utils
+from pyiree.tf.support import tf_utils
+import tensorflow.compat.v2 as tf
+
+FLAGS = flags.FLAGS
+
+# Controls the input signature, call kwargs, input generator and args for a
+# unittest of a tf.math function.
+Config = collections.namedtuple("Config",
+                                ["signature", "kwargs", "generator", "args"])
+# Use old default API for compatibility with Python 3.6.
+Config.__new__.__defaults__ = (None, dict(), None, None)
+
+
+def create_signature_from_args(config: Config) -> Config:
+  """Uses a Config's args to generate an signature for tf.function."""
+  if config.signature is not None:
+    raise ValueError(
+        "Tried to add a signature to a config that already has one.")
+  if config.args is None:
+    raise ValueError(
+        "Tried to generate a signature from a config without any args.")
```

Comment:
These kinds of checks suggest to me that these are really different types. You're not transforming config -> config, you're transforming args -> args + signature, etc. Would a config without a signature be valid? Seems like not

Reply:
Refactored into `Config` and a child `ArgsConfig` that replaces this function.